xray/utils/analyzermanager.go

```diff
@@ -0,0 +1,130 @@
+	analyzerManagerLogFolder = ""
+)
+
+const (
+	analyzerManagerDirName   = "analyzerManagerLogs"
+	jfUserEnvVariable        = "JF_USER"
+	jfPasswordEnvVariable    = "JF_PASS"
+	jfTokenEnvVariable       = "JF_TOKEN"
+	jfPlatformUrlEnvVariable = "JF_PLATFORM_URL"
+	logDirEnvVariable        = "AM_LOG_DIRECTORY"
+	applicabilityScanCommand = "ca"
+)
+
+const (
+	ApplicableStringValue                = "Applicable"
+	NotApplicableStringValue             = "Not Applicable"
+	ApplicabilityUndeterminedStringValue = "Undetermined"
+)
+
+type ExtendedScanResults struct {
+	XrayResults                 []services.ScanResponse
+	ApplicabilityScannerResults map[string]string
+	EntitledForJas              bool
+}
+
+func (e *ExtendedScanResults) GetXrayScanResults() []services.ScanResponse {
+	return e.XrayResults
+}
+
+// AnalyzerManagerInterface represents the analyzer manager executable file that exists locally as a Jfrog dependency.
+// It triggers JAS capabilities by verifying user's entitlements and running the JAS scanners.
+// Analyzer manager input:
+//   - scan command: ca (contextual analysis) / sec (secrets) / iac
+//   - path to configuration file
+//
+// Analyzer manager output:
+//   - sarif file containing the scan results
+type AnalyzerManagerInterface interface {
+	ExistLocally() (bool, error)
+	Exec(string) error
+}
+
+type AnalyzerManager struct {
+	analyzerManagerFullPath string
+}
+
+func (am *AnalyzerManager) ExistLocally() (bool, error) {
+	analyzerManagerPath, err := getAnalyzerManagerAbsolutePath()
+	if err != nil {
+		return false, err
+	}
+	am.analyzerManagerFullPath = analyzerManagerFilePath
+	return fileutils.IsFileExists(analyzerManagerPath, false)
+}
+
+func (am *AnalyzerManager) Exec(configFile string) error {
+	return exec.Command(am.analyzerManagerFullPath, applicabilityScanCommand, configFile).Run()
+}
+
+func CreateAnalyzerManagerLogDir() error {
+	logDir, err := coreutils.CreateDirInJfrogHome(filepath.Join(coreutils.JfrogLogsDirName, analyzerManagerDirName))
+	if err != nil {
+		return err
+	}
+	analyzerManagerLogFolder = logDir
+	return nil
+}
+
+func getAnalyzerManagerAbsolutePath() (string, error) {
+	jfrogDir, err := config.GetJfrogDependenciesPath()
+	if err != nil {
+		return "", err
+	}
+	analyzerManager := analyzerManagerFilePath
+	if coreutils.IsWindows() {
+		analyzerManagerFilePath += ".exe"
+	}
+	return filepath.Join(jfrogDir, analyzerManager), nil
+}
+
+func RemoveDuplicateValues(stringSlice []string) []string {
+	keys := make(map[string]bool)
+	finalSlice := []string{}
+	for _, entry := range stringSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			finalSlice = append(finalSlice, entry)
+		}
+	}
+	return finalSlice
+}
+
+func SetAnalyzerManagerEnvVariables(serverDetails *config.ServerDetails) error {
+	if serverDetails == nil {
+		return errors.New("cant get xray server details")
+	}
+	err := os.Setenv(jfUserEnvVariable, serverDetails.User)
+	if err != nil {
+		return err
+	}
```

Comment:
Can be simplified to - 
```
	if err := os.Setenv(jfUserEnvVariable, serverDetails.User); err != nil {
		return err
	}
```
The same goes for the other if statements in this function.